src/doc/reference.md

```diff
@@ -3652,47 +3652,71 @@ sites are:
 
 * `let` statements where an explicit type is given.
 
-    In `let _: U = e;`, `e` is coerced to have type `U`.
+   For example, `128` is coerced to have type `i8` in the following:
+
+   ```rust
+   let _: i8 = 128;
+   ```
```

Comment:
I don't think this is a good example? 128 is outside the range of `i8`, and I wouldn't think of this as coercion anyway, since the literal's type should be inferred to be `i8`.